app/labor/markdown_parser.rb

```diff
@@ -4,9 +4,10 @@ class MarkdownParser
 
   WORDS_READ_PER_MINUTE = 275.0
 
-  def initialize(content, source: {})
+  def initialize(content, source: nil, user: nil)
```

Comment:
I'm thinking with this setup the `MarkdownParser` as a whole can have come "contextual awareness". It will be able to understand the context that it's parsing markdown for. Context isn't always necessary, so we default to `nil`.